Player/Scripts/Player.gd

```diff
@@ -24,16 +24,21 @@ enum {
 	ATTACK
 }
 
-var state = IDLE
-var flip = false
+var state:= IDLE
+var flip:= false
+export (bool) var attack 
 
-onready var sprite = $Body
-onready var anim = $AnimationPlayer
-onready var hold = $Hold
+onready var sprite := $Body
+onready var anim := $AnimationPlayer
+onready var hold := $Hold
 
-signal collecItem()
+signal collectItem
+signal attack(state)
```

Comment:
New signals for telling the inventory to update the sprites and for the held weapon to activate and deactivate the weapon hurtBox according to player animation respectively.